Add readme for the copy handler component (#25342)
WordPress/gutenberg: packages/block-editor/src/components/copy-handler/README.md

# Copy Handler

> The `CopyHandler` component handles the behaviors of the editor when the user copies or cuts a block or a selection of blocks.

Reviewer: Maybe something more like:

> The `CopyHandler` component handles the `copy`, `cut` and `paste` events of its children blocks. It handles multiple selection of blocks as well.

> Concretely, it handles the display of success messages and takes care of copying the block to the clipboard using the [raw handling API](https://github.com/WordPress/gutenberg/tree/master/packages/blocks/src/api/raw-handling).

Reviewer: @mcsf can you please take a look here?

> Concretely, it handles the display of success messages and takes care of copying the block to the clipboard using the [raw handling API](https://github.com/WordPress/gutenberg/tree/master/packages/blocks/src/api/raw-handling).

Reviewer: >takes care of copying the block

Can you change this to `block/blocks`?

Author: You mean change "takes care of copying the block" to "takes care of copying the block/blocks"?

> Concretely, it handles the display of success messages and takes care of copying the block to the clipboard using the [raw handling API](https://github.com/WordPress/gutenberg/tree/master/packages/blocks/src/api/raw-handling).

Reviewer 2: > copying the block to the clipboard using the raw handling API

Actually, the raw handling API is only used with _paste_ events (`pasteHandler`).

In contrast, cut/copy actions require what is essentially the opposite of the raw handling API: the `serialize` function, which outputs HTML augmented with the HTML-comment demarcations to denote blocks.

Both `pasteHandler` and `serialize` come from the `@wordpress/blocks` module, which is sometimes known as the _blocks _API_.

> ![Copy cut behaviours](https://make.wordpress.org/core/files/2020/09/copy-cut-behaviour.gif)

Reviewer 2: Two notes:

* The GIF itself doesn't clearly show what is meant to happen. Perhaps [this one from the original pull request](https://user-images.githubusercontent.com/150562/81698101-6e341d80-945d-11ea-9bfb-b20781f55033.gif) would be better.

* _Copy cut behaviours_ should be _Copy/cut behaviours_ or _Copy and cut behaviours_.

## Table of contents

1. [Development guidelines](#development-guidelines)
2. [Related components](#related-components)

## Development guidelines

### Usage

> Adds the copy handler functionnalities to the editor.

Reviewer 2: ```suggestion
Adds block-level copy-and-paste support to the editor.
```

Reviewer 2: It's important that this distinction be clear, as `CopyHandler` doesn't handle (and isn't required for) copying and pasting of fragments of content within a block. Those will be natively handled (e.g. selecting a piece of the block UI and copying, or selecting a fragment of RichText and copying).

```jsx
import { BlockList, CopyHandler } from '@wordpress/block-editor';

const MyCopyHandler = () => (
	<CopyHandler>
		<BlockList />
	</CopyHandler>
);
```

> _Note:_ The `CopyHandler` only catch cut/copy events coming from its props.children. In this example, the child is the `BlockList` component.

Reviewer: The events that are handled are three: cut, copy, paste. I think this should be just a part of the main description.

> _Note:_ The `CopyHandler` only catch cut, copy and paste events coming from its props.children. In this example, the child is the `BlockList` component.

Reviewer: >only catch cut,

Change to `catches`

> _Note:_ The `CopyHandler` only catches cut, copy and paste events coming from its props.children. In this example, the child is the `BlockList` component.

Reviewer 2: `CopyHandler` also only intercepts such events when there is a block-editing context and at least one block is currently selected.

### Props

### `children`

> -   **Type:** `Object`

Reviewer: ```suggestion
-   Type: `Element`
```

> An object that contains the component children.

Reviewer: ```suggestion
Children are passed as children of `CopyHandler `.
```

> Children are passed as children of `CopyHandler`.

Reviewer 2: This sentence is a bit redundant. :) 

```suggestion
The elements to be rendered and whose `cut`, `copy` and `paste` events we'd like to intercept.
```
